ggml-alloc.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "ggml.h"
+
+#ifdef  __cplusplus
+extern "C" {
+#endif
+
+
+GGML_API struct ggml_allocator * ggml_allocator_new(void * data, size_t size, size_t alignment);
+GGML_API struct ggml_allocator * ggml_allocator_new_measure(size_t alignment);
+GGML_API void ggml_allocator_free(struct ggml_allocator * alloc);
+GGML_API bool ggml_allocator_is_measure(struct ggml_allocator * alloc);
+GGML_API void ggml_allocator_reset(struct ggml_allocator * alloc);
+GGML_API void ggml_allocator_alloc_tensor(struct ggml_allocator * alloc, struct ggml_tensor * tensor);
+GGML_API size_t ggml_allocator_alloc_graph_tensors(struct ggml_allocator * alloc, struct ggml_cgraph * graph);
```

Comment:
Maybe go with a shorter prefix: `ggml_alloc`

Reply:
`ggml_alloc_alloc_tensor` or just `ggml_alloc_tensor`?

Reply:
How about `ggml_alloc_new_tensor`?

Reply:
I went with the "allocr" prefix so that the difference between "allocator" and "allocate" is a bit clearer.